### PR TITLE
Remove deprecated code segments

### PR DESCRIPTION
**Pull request by Thomas--S**
The removed things here seem to be very old, I don't think that they are needed any longer.

PR was quickly tested in singleplayers, seems to work fine.

### CODE REVIEW

File: src/inventory.cpp [hunk 1]
```diff
@@ -134,7 +117,6 @@ void ItemStack::deSerialize(std::istream &is, IItemDefManager *itemdef)
 			throw SerializationError("Too large material number");
 		// Convert old id to name
```
Comment (sfan5) [inline src/inventory.cpp]: this whole section is rendered useless as `legacy_nimap` is not populated

File: src/inventory.cpp [hunk 2]
```diff
@@ -107,14 +94,10 @@ void ItemStack::deSerialize(std::istream &is, IItemDefManager *itemdef)
 		is>>material;
 		u16 materialcount;
 		is>>materialcount;
-		// Convert old materials
```
Comment (sfan5) [inline src/inventory.cpp]: same here
pretty much the full first two `if` conditions are useless

File: src/inventory.cpp (hunk 1 above)
Comment (Thomas--S) [inline src/inventory.cpp]: So, you think that the whole `else if(name == "MaterialItem2")` branch can be removed?

**Comment by nerzhul** [comment]
thanks, i was thinking about this after release. I'm okay with this, but for next release (maybe on monday :) )

**Comment by Thomas--S** [comment]
Should I already squash both commits or should I wait for the second approval? (And thanks for your review, nerzhul!)

**Comment by nerzhul** [comment]
you don't need to squash we will do it at merge (github permits to do it)

File: src/inventory.cpp [hunk 3]
```diff
@@ -99,140 +86,42 @@ void ItemStack::deSerialize(std::istream &is, IItemDefManager *itemdef)
...
-			// Read metadata
-			metadata.deSerialize(is);
+		// Read metadata
+		metadata.deSerialize(is);
 
-			// In case fields are added after metadata, skip space here:
-			//std::getline(is, tmp, ' ');
-			//if(!tmp.empty())
-			//	throw SerializationError("Unexpected text after metadata");
+		// In case fields are added after metadata, skip space here:
```
Comment (nerzhul) [inline src/inventory.cpp]: you can remove thise comment i think

File: src/inventory.cpp [hunk 4]
```diff
@@ -99,140 +86,42 @@ void ItemStack::deSerialize(std::istream &is, IItemDefManager *itemdef)
...
-		wear = stoi(trim(fnd.next("")));
-	}
-	else
-	{
-		do  // This loop is just to allow "break;"
+
+		// Read the count
+		std::string count_str;
+		std::getline(is, count_str, ' ');
+		if(count_str.empty())
```
Comment (nerzhul) [inline src/inventory.cpp]: missing space & brace on wrong line

File: src/inventory.cpp [hunk 5]
```diff
@@ -99,140 +86,42 @@ void ItemStack::deSerialize(std::istream &is, IItemDefManager *itemdef)
...
-			std::string wear_str;
-			std::getline(is, wear_str, ' ');
-			if(wear_str.empty())
-				break;
-			else
-				wear = stoi(wear_str);
+		// Read the wear
+		std::string wear_str;
+		std::getline(is, wear_str, ' ');
+		if(wear_str.empty())
```
Comment (nerzhul) [inline src/inventory.cpp]: space

File: src/inventory.cpp [hunk 6]
```diff
@@ -99,140 +86,42 @@ void ItemStack::deSerialize(std::istream &is, IItemDefManager *itemdef)
...
-			count = 1;
-	}
-	else if(name == "MBOItem")
-	{
-		// Obsoleted on 2011-10-14
-		throw SerializationError("MBOItem not supported anymore");
-	}
-	else if(name == "tool" || name == "ToolItem")
+	do  // This loop is just to allow "break;"
 	{
```
Comment (nerzhul) [inline src/inventory.cpp]: brace

File: src/inventory.cpp [hunk 7]
```diff
@@ -99,140 +86,35 @@ void ItemStack::deSerialize(std::istream &is, IItemDefManager *itemdef)
...
-				wear = stoi(wear_str);
+		// Read the count
+		std::string count_str;
+		std::getline(is, count_str, ' ');
+		if (count_str.empty()) {
+			count = 1;
+			break;
+		}
+		else
+			count = stoi(count_str);
```
Comment (ShadowNinja) [inline src/inventory.cpp]: The else clause should have braces and the else should be on the same line as the braces.

**Comment by ShadowNinja** [comment]
If we're removing support for MapBlock versions <22, support should be dropped everywhere, not just in the MapNode code.

File: src/inventory.cpp [hunk 8]
```diff
@@ -24,7 +24,6 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include "log.h"
 #include "itemdef.h"
 #include "util/strfnd.h"
-#include "content_mapnode.h" // For loading legacy MaterialItems
 #include "nameidmapping.h" // For loading legacy MaterialItems
```
Comment (sfan5) [inline src/inventory.cpp]: can this include be removed?

**Comment by Thomas--S** [comment]
Thanks for merging!

**Pull request merged**